scripts/generator/graphql_generator/csharp/SDK/Android/AndroidWebCheckout.cs.erb

```diff
@@ -0,0 +1,41 @@
+namespace <%= namespace %>.SDK.Android {
+    #if UNITY_ANDROID
+    using System;
+    using <%= namespace %>.SDK;
+
+    class AndroidWebCheckout: WebCheckout {
+
+        private ShopifyClient _client;
+        private Cart _cart;
+
+        protected override ShopifyClient Client {
+            get {
+                return _client;
+            }
+        }
+
+        protected override Cart Cart {
+            get {
+                return _cart;
+            }
+        }
+
+        private const string UnityPlayerClassName = "com.unity3d.player.UnityPlayer";
+        private const string WebCheckoutSessionClassName = "com.shopify.unity.buy.WebCheckoutSession";
+
+        public AndroidWebCheckout(Cart cart, ShopifyClient client) {
+            _cart = cart;
+            _client = client;
+        }
+
+        public void Checkout(string checkoutURL, CheckoutSuccessCallback success, CheckoutCancelCallback cancelled, CheckoutFailureCallback failure) {
+            #if !SHOPIFY_MONO_UNIT_TEST
+            using(var webCheckoutSession = new AndroidJavaObject(WebCheckoutSessionClassName)) {
```

Comment:
One thing I would look into is exactly when this object will be destroyed. Does it get release when we fall out of the `Checkout` scope? Do we want to maintain a handle to the Java object while `AndroidWebCheckout` is still around?